research/src/components/layout.js

```diff
@@ -56,12 +56,11 @@ const components = {
 }
 
 const Layout = ({ children, pageContext }) => {
-  const { frontmatter } = pageContext || {}
+  const [opened, setOpen] = React.useState(false)
+  const onToggleMenu = React.useCallback(() => setOpen((opened) => !opened), [setOpen])
 
-  const ContentWrapper =
-    frontmatter && frontmatter.path && frontmatter.path.startsWith('/components/')
-      ? ComponentLayout
-      : ({ children }) => <>{children}</>
```

Comment:
This line was effectively creating a new component on each render of Layout, causing React to rebuild the page each time the menu was toggled. Fixed it by just rendering children directly instead of wrapping them in a function.